tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,530 @@
+        """
+        # GIVEN the manifest validation endpoint and parameters
+        url = f"{testing_config.schematic_api_server_url}/v1/model/validate"
+        params = {
+            "schema_url": EXAMPLE_SCHEMA_URL,
+            "data_type": input_data_type,
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+        }
+
+        # AND a valid file
+        file_path = helpers.get_data_path(input_file_name)
+
+        # WHEN we make a POST request to validate the file
+        response = (
+            requests.post(
+                url,
+                params=params,
+                files={"file_name": open(file_path, "rb")},
+                headers=request_headers,
+            )
+            if testing_config.use_deployed_schematic_api_server
+            else flask_client.post(
+                url,
+                headers=request_headers,
+                query_string=params,
+                data={"file_name": open(file_path, "rb")},
+            )
+        )
+
+        # THEN we expect a successful response
+        assert (
+            response.status_code == 200
+        ), f"Got status code: {response.status_code}. Expected '200'."
+
+        # AND with expected keys in the json
+        response_json = (
+            response.json()
+            if testing_config.use_deployed_schematic_api_server
+            else response.json
+        )
+        assert "warnings" in response_json.keys()
+        assert "errors" in response_json.keys()
+
+        # AND with no expected errors
+        assert len(response_json.get("errors")) == 0
+
+    @pytest.mark.local_or_remote_api
+    @pytest.mark.parametrize(
+        ("input_data_type", "input_file_name"),
+        [
+            (
+                "Patient",
+                "mock_manifests/TestManifestValidation_test_patient_manifest_invalid.csv",
+            ),
+        ],
+    )
+    def test_manifest_validation_basic_invalid(
+        self,
+        input_data_type: str,
+        input_file_name: str,
+        flask_client: FlaskClient,
+        request_headers: Dict[str, str],
+        testing_config: ConfigurationForTesting,
+        helpers: Helpers,
+    ) -> None:
+        """
+        Test that the manifest validation API returns errors when an invalid manifest is provided.
+        """
+        # GIVEN the manifest validation endpoint and parameters
+        url = f"{testing_config.schematic_api_server_url}/v1/model/validate"
+        params = {
+            "schema_url": EXAMPLE_SCHEMA_URL,
+            "data_type": input_data_type,
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+        }
+
+        # AND an invalid file
+        file = helpers.get_data_path(input_file_name)
+
+        # WHEN we make a POST request to validate the file
+        response = (
+            requests.post(
+                url,
+                params=params,
+                files={"file_name": open(file, "rb")},
```

Comment:
Commenting here so we don't run into branch conflicts: These files aren't closed at the end of their test. Could this lead to resource issues if multiple branches are running CI/CD tests with open files that are never closed within their environments? @BryanFauble

Reply:
I see the concern, this is how other tests were setup so I used this for consistency.

> Could this lead to resource issues if multiple branches are running CI/CD tests with open files that are never closed within their environments?

No. The file system is not shared between GH runners.

Reply:
I see, thanks! NPB :) but I think this would still pose an issue as we continue adding tests to `test_manifest_validation`, since all these tests are executed sequentially in the same GH runner. I can just make a note of it when I'm working on [this ticket](https://sagebionetworks.jira.com/browse/FDS-2510).